_posts/2025-05-09-query-insights-live-queries.md

```diff
@@ -0,0 +1,141 @@
+
+While historical analysis of top N queries helps you understand trends and past issues, there are times when you need to see exactly what's happening *right now*. Is a particular query taking too long? Is a sudden surge in resource consumption linked to specific search activity? You can gain this immediate insight using live queries.
+
+## What are Live Queries?
+
+Introduced in OpenSearch 3.0, live queries allow you to look into the current state of your cluster's search workload. The Live Queries API retrieves a list of search queries that are currently running across your cluster or on specific nodes. This capability provides real-time visibility into query behavior, which is especially useful for performing the following tasks:
+
+*   **Identifying problematic queries**: Quickly spot queries that are running for an unexpectedly long time.
+*   **Debugging resource hogs**: Pinpoint searches consuming significant CPU or memory *at this very moment*.
+*   **Understanding immediate cluster load**: Get a snapshot of current search activity impacting your cluster.
+
+The API returns key details for each live query, including its source, search type, the indexes involved, the node ID it's running on, its start time, current latency, and resource usage (on the coordinator node) up to that point.
+
+## How live queries work
+
+You can access live query information using the following REST API endpoint:
+
+```json
+GET /_insights/live_queries
+```
+
+By default, the API returns a list of currently executing search queries, sorted by `latency` in descending order.
+
+### Key information returned
+
+For each live query, you'll receive a rich set of data:
+
+*   `timestamp`: The time the query task started, in milliseconds since epoch.
+*   `id`: The unique search task ID.
+*   `description`: Details about the query, including target indexes, search type, and the query source itself (if `verbose` is true).
+*   `node_id`: The ID of the coordinator node on which the query task is running.
+*   `measurements`: An object containing performance metrics gathered so far:
+    *   `latency`: The current running time, in nanoseconds.
+    *   `cpu`: The CPU time consumed so far, in nanoseconds.
+    *   `memory`: The amount of heap memory used so far, in bytes.
+
+## Getting started with live queries
+
+Interacting with the Live Queries API is straightforward. The API supports the following operations.
+
+### Basic request
+
+To get a list of currently running queries sorted by latency, send the following request:
+
+```json
+GET /_insights/live_queries
+```
+
+### Customizing your view using query parameters
```

Comment:
I think it's better to provide a brief overview rather than a parameter table. We can link to the documentation. I'll make a suggestion.